File: src/main/java/net/technearts/Member.java
package net.technearts;

import static org.apache.commons.lang3.StringUtils.isNotBlank;

import lombok.Data;

@Data
public class Member {
  private String title;
  private String column;
  private String converter;
  private String property;
  private String mappedBy;
  private boolean mapped = true;

  public boolean isTitleBased() {
    return isNotBlank(title);
  }

  public boolean isReferenceBased() {
    return isNotBlank(title) || isNotBlank(column);
  }
}

File: src/main/kotlin/net/technearts/Member.kt
package net.technearts

import org.apache.commons.lang3.StringUtils.isNotBlank
class Member {

    var title: String? = null
    var column: String? = null
    var converterName: String? = null
    var converter: ((String) -> Any)? = null
    var property: String? = null
    var mappedBy: String? = null
    var isMapped = true

    val isTitleBased: Boolean
        get() = isNotBlank(title)

    val isReferenceBased: Boolean
        get() = isNotBlank(title) || isNotBlank(column)
}
